### PR TITLE
MoarVM, nqp, rakudo: update to 2021.05

### PR DESCRIPTION
**Pull request by mryan**
###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/nqp/Portfile [hunk 1]
```diff
@@ -43,3 +43,6 @@ universal_variant   no
 # configure.universal_args-delete --disable-dependency-tracking
 
 test.run            yes
+
+# we need to avoid '-' in version name, else $version-RC1 is considered newer than the actual release
+livecheck.regex     {archive/([0-9.]+).tar.gz}
```
Comment (reneeotten) [inline lang/nqp/Portfile:48]: this should just be the usual `github.livecheck.regex  {([0-9.]+)}` and no need for a comment, this is quite common

File: lang/MoarVM/Portfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ PortSystem          1.0
 PortGroup           compiler_blacklist_versions 1.0
 
 name                MoarVM
-version             2021.03
+version             2021.02
```
Comment (reneeotten) [inline lang/MoarVM/Portfile:8]: why are you going down in version number here and in the other ports?!

Also since you're touching three different ports, there should be ideally three separate commits one for each update.

**Comment by mryan** [comment]
Sorry about that - I don't really know how I managed to do that.
I'll close this one and resubmit - it's supposed to be an upgrade to 21.05

**Comment by reneeotten** [comment]
> Sorry about that - I don't really know how I managed to do that.
> 
> I'll close this one and resubmit - it's supposed to be an upgrade to 21.05

No worries! There was no need to close this PR though, ideally you would just have amended your commits and force-pushed to your branch to update this PR.
